spec.md

```diff
@@ -177,18 +181,37 @@ The following attributes are REQUIRED to be present in all CloudEvents:
 #### source
 
 - Type: `URI-reference`
-- Description: This describes the event producer. Often this will include
-  information such as the type of the event source, the organization publishing
-  the event, the process that produced the event, and some unique identifiers.
-  The exact syntax and semantics behind the data encoded in the URI is event
-  producer defined.
+- Description: Identifies the context in which an event
+  happened. Often this will include information such as the type of
+  the event source, the organization publishing the event or the
+  process that produced the event. The exact syntax and semantics
+  behind the data encoded in the URI is defined by the event producer.
+
+  Producers MUST ensure that `source` + `id` is unique for each
+  distinct event.
+
+  An application MAY assign a unique `source` to each distinct
+  producer, which makes it easy to produce unique IDs since no other
+  producer will have the same source. The application MAY use UUIDs,
+  URNs, DNS authorities or an application-specific scheme to create
+  unique `source` identifiers.
+
+  A source MAY include more than one producer. In that case the
```

Comment:
is the word "include" the right word? Perhaps "use" or "leverage"?  Not a big deal since I think I get the point, but "include" might sound like we're getting include impl details and suggesting that a producer is a sub-component of a source - which it may not be.